From a45e354a939c4db0e223c3536e6df471b2bc0f61 Mon Sep 17 00:00:00 2001
From: Patrick Thier <pthier@chromium.org>
Date: Wed, 8 Dec 2021 11:31:15 +0000
Subject: [PATCH] Activate v8_include_receiver_in_argc unconditionally

All uses of the build flag will be removed in a follow-up CL.

Bug: v8:11112
Change-Id: Ifc3d9ed5d642646edeba51aa6d60924796d0f788
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3322769
Auto-Submit: Patrick Thier <pthier@chromium.org>
Reviewed-by: Camillo Bruni <cbruni@chromium.org>
Commit-Queue: Camillo Bruni <cbruni@chromium.org>
Cr-Commit-Position: refs/heads/main@{#78294}
---
 BUILD.gn | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index ee6d007e723..0952eb37e20 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -352,12 +352,7 @@ declare_args() {
   # parameter count of function with JS linkage.
   # TODO(v8:11112): Remove once all architectures support the flag and it is
   # enabled unconditionally.
-  v8_include_receiver_in_argc =
-      v8_current_cpu == "x86" || v8_current_cpu == "x64" ||
-      v8_current_cpu == "arm" || v8_current_cpu == "arm64" ||
-      v8_current_cpu == "mips64el" || v8_current_cpu == "mipsel" ||
-      v8_current_cpu == "loong64" || v8_current_cpu == "riscv64" ||
-      v8_current_cpu == "ppc64" || v8_current_cpu == "s390x"
+  v8_include_receiver_in_argc = true
 }
 
 # Derived defaults.
-- 
2.35.1

